src/core/blockchain/celo/stats.ts

```diff
@@ -1,275 +1,59 @@
-import {
-    GenericStats,
-    IStatValueType,
-    AccountStats,
-    IValidator,
-    CardActionType
-} from '../types/stats';
+import { GenericStats, AccountStats, IStatValueType } from '../types/stats';
 import { Client } from './client';
-import { Blockchain } from '../types/blockchain';
+import { Contracts } from './config';
+import { IAccountState, ITokenState } from '../../../redux/wallets/state';
+import { ApiClient } from '../../utils/api-client/api-client';
+import BigNumber from 'bignumber.js';
+import { Blockchain } from '../types';
 
 export class Stats extends GenericStats<Client> {
-    public async getAccountDelegateStats(): Promise<AccountStats> {
-        return {
-            topStats: topStatsValues,
-            chartStats: chartStatsValues,
-            secondaryStats: secondaryStatsValues
-        };
+    public async getAccountDelegateStats(
+        account: IAccountState,
+        token: ITokenState
+    ): Promise<AccountStats> {
+        const data = await new ApiClient().validators.getAccountDelegateStats(
+            account,
+            this.client.chainId.toString()
+        );
+
+        if (data === undefined) {
```

Comment:
why don't we do this in API ? it would be better, and it solves this kind of issue everywhere.